src/cdk/v2/destinations/koddi/utils.js

```diff
@@ -0,0 +1,116 @@
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const { EVENT_NAMES, IMPRESSIONS_CONFIG, CLICKS_CONFIG, CONVERSIONS_CONFIG } = require('./config');
+const {
+  constructPayload,
+  defaultRequestConfig,
+  toUnixTimestamp,
+  stripTrailingSlash,
+} = require('../../../../v0/util');
+
+const validateBidders = (bidders) => {
+  if (!Array.isArray(bidders)) {
+    throw new InstrumentationError('properties.bidders should be an array of objects. Aborting.');
+  }
+  if (bidders.length === 0) {
+    throw new InstrumentationError(
+      'properties.bidders should contains at least one bidder. Aborting.',
+    );
+  }
+  bidders.forEach((bidder) => {
+    if (!(bidder.bidder || bidder.alternate_bidder)) {
+      throw new InstrumentationError('bidder or alternate_bidder is not present. Aborting.');
+    }
+    if (!bidder.count) {
+      throw new InstrumentationError('count is not present. Aborting.');
+    }
+    if (!bidder.base_price) {
+      throw new InstrumentationError('base_price is not present. Aborting.');
+    }
+  });
+};
+
+/**
+ * This function constructs payloads based upon mappingConfig for all calls.
+ * @param {*} eventName
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const constructFullPayload = (eventName, message, Config) => {
+  let payload;
+  switch (eventName) {
+    case EVENT_NAMES.IMPRESSIONS:
+      payload = constructPayload(message, IMPRESSIONS_CONFIG);
+      payload.clientName = Config.clientName;
+      break;
+    case EVENT_NAMES.CLICKS:
+      payload = constructPayload(message, CLICKS_CONFIG);
+      payload.clientName = Config.clientName;
+      if (!Config.testVersionOverride) {
+        payload.testVersionOverride = null;
+      }
+      if (!Config.overrides) {
+        payload.overrides = null;
+      }
+      break;
+    case EVENT_NAMES.CONVERSIONS:
+      payload = constructPayload(message, CONVERSIONS_CONFIG);
+      payload.client_name = Config.clientName;
+      payload.unixtime = toUnixTimestamp(payload.unixtime);
```

Comment:
does koddi accepts timestamp in ms or sec?

Reply:
It will be in second.
https://developers.koddi.com/reference/conversions-1

<img width="1415" alt="Screenshot 2024-05-20 at 3 37 07 PM" src="https://github.com/rudderlabs/rudder-transformer/assets/144022547/121907fd-46fe-4676-b41f-7a971c5f926d">